tests/operators/test_agnostic_merge.py

```diff
@@ -0,0 +1,177 @@
+
+
+@pytest.fixture
+def merge_parameters(request, sql_server):
+    mode = request.param
+    if mode == "single":
+        return (
+            {
+                "merge_keys": merge_keys(sql_server, mode),
+                "merge_columns": ["list"],
+                "target_columns": ["list"],
+                "conflict_strategy": "ignore",
+            },
+            mode,
+        )
+    elif mode == "multi":
+        return (
+            {
+                "merge_keys": merge_keys(sql_server, mode),
+                "merge_columns": ["list", "sell"],
+                "target_columns": ["list", "sell"],
+                "conflict_strategy": "ignore",
+            },
+            mode,
+        )
+    elif mode == "mixed":
+        return (
+            {
+                "merge_keys": merge_keys(sql_server, mode),
+                "target_columns": ["list", "sell", "taxes"],
+                "merge_columns": ["list", "sell", "age"],
+                "conflict_strategy": "update",
+            },
+            mode,
+        )
+
+
+@aql.transform
+def do_a_thing(input_table: Table):
+    return "SELECT * FROM {{input_table}}"
+
+
+@aql.run_raw_sql
+def add_constraint(table: Table, columns):
+    if table.conn_type == "sqlite":
+        return (
+            "CREATE UNIQUE INDEX unique_index ON {{table}}" + f"({','.join(columns)})"
+        )
+    elif table.conn_type == "bigquery":
+        return ""
+    return (
+        "ALTER TABLE {{table}} ADD CONSTRAINT airflow UNIQUE"
+        + f" ({','.join(columns)})"
+    )
+
+
+@adf
+def validate_results(df: pd.DataFrame, mode, sql_type):
+    # make columns lower and reverse due to snowflake defaulting to uppercase
+    # Also reverse because BQ and snowflake seem to reverse row order
+    if sql_type in ["snowflake", "bigquery"]:
+        df.columns = df.columns.str.lower()
+        df = df.iloc[::-1]
+
+    def set_compare(l1, l2):
+        return set(l1) == set(l2)
+
+    if mode == "single":
+        assert set_compare(df.age.to_list()[:-1], [60.0, 12.0, 41.0, 22.0])
+        assert set_compare(df.taxes.to_list()[:-1], [3167.0, 4033.0, 1471.0, 3204.0])
+        assert set_compare(df.taxes.to_list()[:-1], [3167.0, 4033.0, 1471.0, 3204.0])
+        assert set_compare(df.list.to_list(), [160, 180, 132, 140, 240])
+        assert set_compare(df.sell.to_list()[:-1], [142, 175, 129, 138])
+    elif mode == "multi":
+        assert set_compare(df.age.to_list()[:-1], [60.0, 12.0, 41.0, 22.0])
+        assert set_compare(df.taxes.to_list()[:-1], [3167.0, 4033.0, 1471.0, 3204.0])
+        assert set_compare(df.taxes.to_list()[:-1], [3167.0, 4033.0, 1471.0, 3204.0])
+        assert set_compare(df.list.to_list(), [160, 180, 132, 140, 240])
+        assert set_compare(df.sell.to_list()[:-1], [142, 175, 129, 138])
+    elif mode == "mixed":
+        assert df.taxes.to_list() == [1, 1, 1, 1, 1]
+        assert set_compare(df.age.to_list()[:-1], [60.0, 12.0, 41.0, 22.0])
+
+
+@task_group
```

Comment:
That's a neat way of testing them! :love_you_gesture:

Reply:
Thanks @tatiana! I'd love to eventually set up these task_group functions to be something we can mix and match for larger integration tests!